src/logic/calculate.js

```diff
@@ -0,0 +1,23 @@
+import operate from './operate';
+
+const calculate = (calcObj, btnName) => {
+  const { total, next, operation } = calcObj;
+
+  if (btnName === '+/-') {
+    if (total) {
+      return total * -1;
+    }
+    return next * -1;
+  }
+
+  if (btnName === '%') {
+    if (total) {
+      return operate(total, 100, '/');
+    }
+    return operate(next, 100, '/');
+  }
```

Comment:
- [ ] I suggest you put this in your `operate.js` module, and use the `mod` function from `Big.js`.